test/SlowTests/Issues/RavenDB-17597.cs

```diff
@@ -0,0 +1,360 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using FastTests;
+using FastTests.Client;
+using NuGet.ContentModel;
+using Orders;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Indexes.Spatial;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.Indexes;
+using Raven.Client.Documents.Session;
+using Raven.Client.Exceptions.Database;
+using Raven.Client.Exceptions.Documents.Indexes;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Operations;
+using Raven.Server;
+using Raven.Server.Commercial;
+using Raven.Server.Config;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Indexes;
+using Raven.Server.ServerWide.Context;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+using Raven.Client;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_17597 : ClusterTestBase
+    {
+        public RavenDB_17597(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task CopyToIndexDefinitionWorksProperly()
+        {
+            var conf = new IndexConfiguration();
+            conf.Add("key1", "val1");
+            conf.Add("key2", "val2");
+
+            IndexDefinition a = new IndexDefinition()
```

Comment:
you also need to compare with default values `var defaultDefinition = new IndexDefinition()`
I would like all values in `a` to be different from the default ones.

Reply:
BTW, here is a IndexDefinition.Compare method, use it?

Reply:
I wanted to write a test that will check all the fields with reflection, that even if we will add a property to new IndexDefinition class, the test still checks it (with reflection)..
that's why I didn't use  the compare method

Reply:
Use `Compare`, but add a test that ensure that the number of fields it checks and the number of fields on the class are the same.

Reply:
I think this can be done easily by checking number of values in the enum that this Compare returns.

Reply:
Done